### PR TITLE
gc namespace for k8sspark k8sflink

### PR DESCRIPTION
**Pull request by chengjoey**
#### What type of this PR

Add one of the following kinds:
/kind feature

#### What this PR does / why we need it:
gc namespace for k8sspark k8sflink

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/dop/projects/387/issues/all?id=265418&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sInN0YXRlcyI6WzQ0MDIsNzEwNCw3MTA1LDQ0MDMsNDQwNCw3MTA2LDQ0MDYsNDQwNyw0NDEyLDQ1MzgsNDQxMyw0NDE0LDQ0MTUsNDQxNl0sImFzc2lnbmVlSURzIjpbIjEwMDEyMDUiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=680&type=TASK)
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/dop/projects/387/issues/all?id=265633&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sInN0YXRlcyI6WzQ0MDIsNzEwNCw3MTA1LDQ0MDMsNDQwNCw3MTA2LDQ0MDYsNDQwNyw0NDEyLDQ1MzgsNDQxMyw0NDE0LDQ0MTUsNDQxNl0sImFzc2lnbmVlSURzIjpbIjEwMDEyMDUiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=680&type=BUG)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | Support gc namespace for k8sspark k8sflink             |
| 🇨🇳 中文    |实现k8sspark和k8sflink类型的pipeline gc namespace              |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by chengjoey** [comment]
image

File: modules/pipeline/conf/conf.go [hunk 1]
```diff
@@ -112,6 +112,9 @@ type Conf struct {
 	// scheduler executor refresh interval
 	ExecutorRefreshIntervalMinute uint64 `env:"EXECUTOR_REFRESH_INTERVAL_MINUTE" default:"20"`
 	SpecifyImagePullPolicy        string `env:"SPECIFY_IMAGE_PULL_POLICY" default:"IfNotPresent"`
+
+	// k8s namespace
+	EnableSpecifiedK8SNamespace string `env:"ENABLE_SPECIFIED_K8S_NAMESPACE"`
```
Comment (sfwn) [inline modules/pipeline/conf/conf.go:117]: I think `specified_k8s_namespace` should be a request level config.

Comment (sfwn) [inline modules/pipeline/conf/conf.go:117]: use bool

File: modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sflink/flink.go [hunk 2]
```diff
@@ -182,12 +184,56 @@ func (k *K8sFlink) Remove(ctx context.Context, task *spec.PipelineTask) (interfa
...
 	}
 
 	err = k.client.CRClient.Delete(ctx, flinkCluster)
 	if err != nil {
 		return nil, fmt.Errorf("delete flink cluster %s err: %s", bigDataConf.Name, err.Error())
 	}
+
+	// delete namespace after gc flinkcluster
+	namespace := task.Extra.Namespace
+	if conf.EnableSpecifiedK8SNamespace() == "" && !task.Extra.NotPipelineControlledNs {
```
Comment (sfwn) [inline modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sflink/flink.go:197]: bad string comparation.

File: modules/pipeline/conf/conf.go (hunk 1 above)
Comment (sfwn) [reply modules/pipeline/conf/conf.go:117]: who inject this env

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/conf/conf.go (hunk 1 above)
Comment (chengjoey) [reply modules/pipeline/conf/conf.go]: adjusted, only use `task.Extra.NotPipelineControlledNs` 

NotPipelineControlledNs is request level config

**Review by sfwn — APPROVED** [review]

**Pull request merged**